src/main/java/frc/robot/subsystems/vision/Limelight.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.net.PortForwarder;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import frc.robot.subsystems.LoggedSubsystem;
+
+import java.util.Optional;
+import java.util.OptionalDouble;
+
+public class Limelight extends LoggedSubsystem<LimelightLogInputs> {
+    public static Limelight INSTANCE = null;
+
+    private final NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+    private final NetworkTableEntry tx = table.getEntry("tx");
+    private final NetworkTableEntry ty = table.getEntry("ty");
+    private final NetworkTableEntry tv = table.getEntry("tv");
+    private final NetworkTableEntry ts = table.getEntry("ts");
+    private final NetworkTableEntry tid = table.getEntry("tid");
+    private LimelightLogInputs inputs = LimelightLogInputs.getInstance();
+
+    private Limelight() {
+        super(new LimelightLogInputs());
+        inputs = LimelightLogInputs.getInstance();
+        PortForwarder.add(5801, "limelight.local", 5801);
```

Comment:
Do you understand what this line does and why it is required?

_(And on a personal note – please don't take this comment and others I'll write in the season as condescending. My only intent is that you'll learn and it's important to me to see you fully understand the code you're writing ☺️.
[IIRC](https://www.dictionary.com/browse/iirc) you haven't learned networking basics, so I'm asking the question above.)_

Reply:
Gaya told me to put it in the constructor, I know it's used to forward the connection from the raspberry Pi that is used in the limelight.

Reply:
It's more accurate to say "forward the port", not the connection (that's also how it's written in the code).
Why is this important?

Reply:
During competitions, as the routers are configured to work only with the FMS, you are required to forward the stream so you will be able to use it during calibration.
https://docs.limelightvision.io/en/latest/best_practices.html#before-an-event
As a reference

Reply:
I wanted Emma to answer 🙂.
That is correct though, you can't access the UI of Limelight/PhotonVision when connected to the roboRIO with USB.
Forwarding the UI port through the roboRIO enables us to access the UI regardless of the restrictions Barel mentioned.